sqlmesh/core/model/definition.py

```diff
@@ -1805,7 +1806,7 @@ def create_external_model(
     path: Path = Path(),
     defaults: t.Optional[t.Dict[str, t.Any]] = None,
     **kwargs: t.Any,
-) -> Model:
+) -> ExternalModel:
```

Comment:
I wanted to collect the output of `create_external_model` into a typed list. The type needed to be `ExternalModel` and not `Model` so I could access the `gateway` property without the type checker complaining.

So I changed the return type of `create_external_model()` to be `ExternalModel` but it did mean I needed to cast the result of `_create_model`